gnomad_qc/v3/create_release/sanity_checks.py

```diff
@@ -0,0 +1,711 @@
+) -> None:
+    """
+    Compute afresh the sum of annotations for a specified group of annotations, and compare to the annotated version;
+    displays results from checking the sum of the specified annotations in the terminal.
+    Also check that annotations for all expected sample populations are present.
+    :param hl.MatrixTable, hl.Table t: Input MatrixTable or Table.
+    :param List[str] subsets: List of sample subsets.
+    :param List[str] info_metrics: List of metrics in info struct of input Table.
+    :param bool verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks.
+    :param pop_names: Dict with global population names (keys) and population descriptions (values).
+    :return: None
+    :rtype: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+    # Add "" for sum checks on entire callset
+    subsets.append("")
+    # Perform sample sum checks per subset
+    for subset in subsets:
+        pop_names = pops
+        if subset == "hgdp":
+            pop_names = HGDP_POPS
+        if subset == "tgp":
+            pop_names = KG_POPS
+
+        sample_sum_check(t, subset, dict(group=["adj"], pop=pop_names), verbose)
+        sample_sum_check(t, subset, dict(group=["adj"], sex=sexes), verbose)
+        sample_sum_check(
+            t,
+            subset,
+            dict(group=["adj"], pop=list(set(pop_names)), sex=sexes),
+            verbose,
+        )
+
+
+def sex_chr_sanity_checks(
+    t: Union[hl.MatrixTable, hl.Table],
+    info_metrics: List[str],
+    contigs: List[str],
+    verbose: bool,
+) -> None:
+    """
+    Performs sanity checks for annotations on the sex chromosomes.
+    Check:
+        - That metrics for chrY variants in female samples are NA and not 0
+        - That nhomalt counts are equal to female nhomalt counts for all non-PAR chrX variants
+    :param hl.MatrixTable, hl.Table t: Input MatrixTable or Table.
+    :param List[str] info_metrics: List of metrics in info struct of input Table.
+    :param List[str] contigs: List of contigs present in input Table.
+    :param bool verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks.
+    :return: None
+    :rtype: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    female_metrics = [x for x in info_metrics if "-female" in x or "-XX" in x]
+
+    if "chrY" in contigs:
+        logger.info("Check values of female metrics for Y variants are NA:")
+        t_y = hl.filter_intervals(t, [hl.parse_locus_interval("chrY")])
+        metrics_values = {}
+        for metric in female_metrics:
+            metrics_values[metric] = hl.agg.any(hl.is_defined(t_y.info[metric]))
+        output = t_y.aggregate(hl.struct(**metrics_values))
+        for metric, value in dict(output).items():
```

Comment:
Currently in UKBB for this it is 
```
        for metric, values in dict(output).items():
            if values == {None}:
                logger.info(f"PASSED {metric} = {None} check for Y variants")
            else:
                logger.info(f"FAILED Y check: Found {values} in {metric}")
```
So just confirming this is what we want and also a place that would need to change if this function is generalized